mono/metadata/boehm-gc.c

```diff
@@ -92,27 +92,45 @@ void mono_gc_handle_lock () { lock_handles (NULL);}
 #define unlock_handles(handles) mono_os_mutex_unlock (&handle_section)
 void mono_gc_handle_unlock () { unlock_handles (NULL); }
 
-typedef struct {
+/*
+ * The value of HANDLE_COUNT must satisfy two constraints:
+ * 1. It must be a multiple of CHAR_BIT, e.g. HANDLE_COUNT % CHAR_BIT == 0
+ * 2. It must be small enough that sizeof(HandleData) < HANDLE_DATA_ALIGNMENT
+ *
+ * The two values chosen below for 32 & 64 bit, are the largest values for HANDLE_COUNT
+ * to result in the smallest amount of wasted space at the end of the allocation.
+ */
+#if SIZEOF_VOID_P == 4
+#define HANDLE_COUNT 1008
+#define HANDLE_DATA_ALIGNMENT 4096
```

Comment:
I went with 4k and 8k hunks of handles. We could go much larger.

I am also using virtual alloc calls to allocate the memory. On windows, this uses 64k allocation/alignments, so on Windows we are using that much space regardless.

Reply:
Any thoughts on this? Should we use larger granularity? Should I use an standard allocator rather than virtual memory allocations and align the allocation myself?

I tweaked these numbers and didn't see much performance difference. If we want to keep the sizes as is, should I use malloc to avoid wasting memory?